### PR TITLE
fix: tap to dismiss name editing

### PR DESCRIPTION
**Pull request by besilva**
## Thank you for your contribution! 

Fix: tap to dismiss name editing #162  also updated changelog

### Before merging the PR

- [ ] If applicable, does the PR contain enough unit / UI tests?
- [ ] If applicable, did you add a before / after screenshot of the feature?
- [x] Did you update the [CHANGELOG.md](https://github.com/criticalmaps/criticalmaps-ios/blob/master/CHANGELOG.md)?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Changelog for Critical Maps iOS
 - Fix: NavigationBar Colors under iOS 13 
 - Fix: UITableViewHeaderFooterView backgroundColor deprecation warning
 - Fix: Ambiguous auto layout constraints for Settings screen
+- Fix: Manage Friends tap to dismiss on name editing
```
Comment (lennet) [inline CHANGELOG.md]: @mltbnz, @headione  Should we track the friends changes in the changeling? We just need to make sure that they don't end up in the store release notes yet.

File: CriticalMaps.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1273,7 +1273,7 @@
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
 				CURRENT_PROJECT_VERSION = 19;
-				DEVELOPMENT_TEAM = 5YLLXUZBZ2;
+				DEVELOPMENT_TEAM = P33TK2RVAJ;
```
Comment (lennet) [inline CriticalMaps.xcodeproj/project.pbxproj]: Can you reset project.pbxproj? The changes doesn't seem to be related to the bug fix

File: CriticalMass/ManageFriendsViewController.swift [hunk 3]
```diff
@@ -67,7 +69,7 @@ class ManageFriendsViewController: UIViewController, IBConstructable, UITableVie
         navigationController?.pushViewController(viewController, animated: true)
     }
     
-    @objc private func handleTap() {
+    @objc private func handleTap(_ tap: UITapGestureRecognizer) {
```
Comment (lennet) [inline CriticalMass/ManageFriendsViewController.swift]: Let's remove the argument if we don't need it

File: CriticalMass/ManageFriendsViewController.swift [hunk 4]
```diff
@@ -48,8 +48,10 @@ class ManageFriendsViewController: UIViewController, IBConstructable, UITableVie
         tableView.register(viewType: SettingsTableSectionHeader.self)
 
         //TODO: Fix collision with tableview gesture recognizer
-//        let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(handleTap))
-//        view.addGestureRecognizer(tapGestureRecognizer)
+        let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(handleTap(_:)))
+        tapGestureRecognizer.cancelsTouchesInView = false
+        view.addGestureRecognizer(tapGestureRecognizer)
+
```
Comment (lennet) [inline CriticalMass/ManageFriendsViewController.swift]: minor: remove empty the line

File: CriticalMass/ManageFriendsViewController.swift [hunk 5]
```diff
@@ -48,8 +48,10 @@ class ManageFriendsViewController: UIViewController, IBConstructable, UITableVie
         tableView.register(viewType: SettingsTableSectionHeader.self)
 
         //TODO: Fix collision with tableview gesture recognizer
```
Comment (lennet) [inline CriticalMass/ManageFriendsViewController.swift]: We can remove the TODO as you fixed it :)

File: CHANGELOG.md (hunk 1 above)
Comment (besilva) [reply CHANGELOG.md]: makes sense, I will remove it. I forgot it isn't in the master yet.

**Comment by besilva** [comment]
I've fixed the pr all issues now

**Comment by besilva** [comment]
@lennet Do you want me to implement #163 and #164 ?

File: CHANGELOG.md (hunk 1 above)
Comment (lennet) [reply CHANGELOG.md]: 👍. Sorry for the confusion and misleading issue template!

File: CriticalMaps.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -7,9 +7,9 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
-		75147B7A23466E800056BDBF /* LinkResponsiveTextView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 75147B7923466E800056BDBF /* LinkResponsiveTextView.swift */; };
 		3F137FBA2349CD71006B26AB /* SnapshotHelper.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3F137FB92349CD71006B26AB /* SnapshotHelper.swift */; };
 		3F137FBB2349FCBA006B26AB /* CriticalMapsUITests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3F137FA32349CADC006B26AB /* CriticalMapsUITests.swift */; };
+		75147B7A23466E800056BDBF /* LinkResponsiveTextView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 75147B7923466E800056BDBF /* LinkResponsiveTextView.swift */; };
```
Comment (lennet) [inline CriticalMaps.xcodeproj/project.pbxproj]: Did you move the files or where is this change coming from? Nothing too bad, but would be nice to reset the change if possible to minimise the risk of merge conflicts

File: CriticalMaps.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -168,11 +168,11 @@
...
 
 /* Begin PBXFileReference section */
-		75147B7923466E800056BDBF /* LinkResponsiveTextView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = LinkResponsiveTextView.swift; sourceTree = "<group>"; };
 		3F137FA32349CADC006B26AB /* CriticalMapsUITests.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = CriticalMapsUITests.swift; sourceTree = "<group>"; };
 		3F137FA52349CADC006B26AB /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
 		3F137FAF2349CB8C006B26AB /* CriticalMapsUITests.xctest */ = {isa = PBXFileReference; explicitFileType = wrapper.cfbundle; includeInIndex = 0; path = CriticalMapsUITests.xctest; sourceTree = BUILT_PRODUCTS_DIR; };
 		3F137FB92349CD71006B26AB /* SnapshotHelper.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = SnapshotHelper.swift; sourceTree = "<group>"; };
+		75147B7923466E800056BDBF /* LinkResponsiveTextView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = LinkResponsiveTextView.swift; sourceTree = "<group>"; };
```
Comment (lennet) [inline CriticalMaps.xcodeproj/project.pbxproj]: same here

File: CriticalMaps.xcodeproj/project.pbxproj (hunk 6 above)
Comment (besilva) [reply CriticalMaps.xcodeproj/project.pbxproj]: XCode made it by itself

**Comment by lennet** [comment]
> @lennet Do you want me to implement #163 and #164 ? I can do it and open others prs later

That would be awesome!
